IHP/Postgres/TSVector.hs

```diff
@@ -0,0 +1,13 @@
+{-|
+Module: IHP.Postgres.TSVector
+Description: Adds support for the Postgres tsvector type
+Copyright: (c) digitally induced GmbH, 2021
+-}
+module IHP.Postgres.TSVector where
+
+import BasicPrelude
+
+-- | Represents a Postgres tsvector
+--
+-- See https://www.postgresql.org/docs/current/datatype-textsearch.html
+type TSVector = Text
```

Comment:
This needs to be a newtype

Reply:
I don't think a string representation is the right choice here.

This is a valid tsvector: `a:1A fat:2B,4C cat:5D`

The `:` suffix has special meaning here. A good structure would be:

```haskell
data TSVector = TSVector [Lexeme]
data Lexeme = Lexeme { token :: Text, position :: Int, weight :: Text }
```